transaction-service/src/main/java/pl/adamboguszewski/transaction/service/application/TransactionService.java

```diff
@@ -42,8 +42,13 @@ public GetTransactionDto getByTransactionId(UUID id) {
         Gson gsonBuilder = new GsonBuilder().create();
         String dtoJson = gsonBuilder.toJson(dto);
         log.debug("Creating transaction from given dto: " + dtoJson);
-        Optional<Transaction> transaction = Optional.of(repository.save(new Transaction(dto)));
-        log.info("Transaction with id " + dto.getTransactionId() + " created successfully.");
+        Optional<Transaction> transaction = Optional.ofNullable(repository.save(new Transaction(dto)));
+        if(transaction.isPresent()) {
+            log.info("Transaction with id " + dto.getTransactionId() + " created successfully.");
+        }
+        else {
+            log.debug("Could not create transaction with id " + dto.getTransactionId());
+        }
         return transaction;
```

Comment:
We want to return transaction here, not optional of transaction. We have redundant code in our controller. Our controller createTransaction method should look like:
```java
@PostMapping("")
    public ResponseEntity<CreateTransactionResponse> createTransaction(@Valid @RequestBody CreateTransactionRequest request) {
        Transaction transaction = transactionService.createTransaction(CreateTransactionDto.fromRequest(request));
        return new ResponseEntity<>(new CreateTransactionSuccessResponse(transaction.getId()), HttpStatus.CREATED);
    }
```